packages/host/app/components/operator-mode/realm-info-provider.gts

```diff
@@ -0,0 +1,68 @@
+import Component from '@glimmer/component';
+
+import { service } from '@ember/service';
+import { use, resource } from 'ember-resources';
+import { type RealmInfo } from '@cardstack/runtime-common';
+import { TrackedObject } from 'tracked-built-ins';
+import type RealmInfoService from '@cardstack/host/services/realm-info-service';
+
+export interface Signature {
+  Args: { realmUrl?: string; fileUrl?: string };
+  Blocks: {
+    ready: [RealmInfo];
+    error: [Error];
+  };
+}
+
+export default class RealmInfoProvider extends Component<Signature> {
+  @service declare realmInfoService: RealmInfoService;
+
+  @use private realmInfoResource = resource(() => {
+    if (!this.args.fileUrl) {
+      return new TrackedObject({
+        error: null,
+        isLoading: false,
+        value: null,
+        load: () => Promise<void>,
+      });
+    }
+
+    const state: {
+      isLoading: boolean;
+      value: RealmInfo | null;
+      error: Error | undefined;
+      load: () => Promise<void>;
+    } = new TrackedObject({
+      isLoading: true,
+      value: null,
+      error: undefined,
+      load: async () => {
+        state.isLoading = true;
+
+        try {
+          let realmInfo = await this.realmInfoService.fetchRealmInfo({
+            realmUrl: this.args.realmUrl,
+            fileUrl: this.args.fileUrl,
+          });
+
+          state.value = realmInfo;
+        } catch (error: any) {
+          state.error = error;
+        } finally {
+          state.isLoading = false;
+        }
+      },
+    });
+
+    state.load();
+    return state;
+  });
+
+  <template>
+    {{#if this.realmInfoResource.value}}
+      {{yield this.realmInfoResource.value to='ready'}}
+    {{else if this.realmInfoResource.error}}
+      {{yield this.realmInfoResource.error to='error'}}
+    {{/if}}
```

Comment:
I haven’t seen this pattern for resources and named blocks before, nice!

Reply:
I took the idea from how we loaded modules in the old code route, I also think a provider component with a resource looks pretty neat and clean